### PR TITLE
refactor(kotlin): rewrite microservice in Kotlin

### PR DESCRIPTION
**Pull request by maydek1**
- Converted microservice codebase from Java to Kotlin
- Leveraged Kotlin features for cleaner and more concise

### CODE REVIEW

File: src/main/kotlin/com/software/modsen/carservice/config/MapperConfig.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+class MapperConfig {
+    @Bean
+    fun modelMapper(): ModelMapper {
+        val modelMapper = ModelMapper()
+        modelMapper.configuration.matchingStrategy = MatchingStrategies.STRICT
+        return ModelMapper()
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/carservice/config/MapperConfig.kt]: Check scope function .apply() - you can make your code very readable

File: src/main/kotlin/com/software/modsen/carservice/dto/request/CarRequest.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.software.modsen.carservice.dto.request
+
+data class CarRequest(
+        var model: String? = null,
+        var number: String? = null,
+        var color: String? = null
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/carservice/dto/request/CarRequest.kt]: If request, do not make fields nullable if they don't

File: src/main/kotlin/com/software/modsen/carservice/dto/response/CarResponse.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.software.modsen.carservice.dto.response
+
+data class CarResponse(
+        var id:Long? = null,
+        var number: String? = null,
+        var model: String? = null,
+        var color: String? = null
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/carservice/dto/response/CarResponse.kt]: same here

File: src/main/kotlin/com/software/modsen/carservice/dto/response/CarResponseSet.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.software.modsen.carservice.dto.response
+
+data class CarResponseSet(
+        var items: Set<CarResponse>
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/carservice/dto/response/CarResponseSet.kt]: use val

File: src/main/kotlin/com/software/modsen/carservice/model/Car.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+@Entity
+@Table(name = "car")
+data class Car(
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    var id: Long? = null,
+    var number: String? = null,
+    var model: String? = null,
+    var color: String? = null,
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/carservice/model/Car.kt]: Do not specify null value of a field if nullable

File: src/main/kotlin/com/software/modsen/carservice/repository/CarRepository.kt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.software.modsen.carservice.repository
+
+import com.software.modsen.carservice.model.Car
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.stereotype.Repository
+
+@Repository
+interface CarRepository : JpaRepository<Car,Long> {
+    fun existsByNumber(number: String?): Boolean
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/carservice/repository/CarRepository.kt:9]: Do you really want to search by null?

**Review by psidarovich — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/software/modsen/carservice/config/MapperConfig.kt (hunk 1 above)
Comment (maydek1) [reply src/main/kotlin/com/software/modsen/carservice/config/MapperConfig.kt]: Used .apply()

File: src/main/kotlin/com/software/modsen/carservice/dto/request/CarRequest.kt (hunk 2 above)
Comment (maydek1) [reply src/main/kotlin/com/software/modsen/carservice/dto/request/CarRequest.kt]: Changed

File: src/main/kotlin/com/software/modsen/carservice/dto/response/CarResponseSet.kt (hunk 4 above)
Comment (maydek1) [reply src/main/kotlin/com/software/modsen/carservice/dto/response/CarResponseSet.kt]: used

**Review by psidarovich — APPROVED** [review]

**Pull request merged**